mcm/model_argument_transform.py

```diff
@@ -0,0 +1,112 @@
+from copy import copy
+from typing import Callable, Dict, Iterable, List, Tuple
+import re
+
+import numpy as np
+
+
+categorical_param_name_pattern = r"^(?P<propname>[^\[]+)\[T\.(?P<matchingval>[^\]]+)\]"
+categorical_param_name_regex = re.compile(categorical_param_name_pattern)
+
+
+class IndicatorTransform:
+    """
+    Transform that returns 1 if value matches; else 0.
+
+    matching_value should be either a primitive or a `copy`-able container of primitives.
+    """
+
+    def __init__(self, matching_value):
+        self._matching_value = copy(matching_value)
+
+    @property
+    def matching_value(self):
+        return self._matching_value
+
+    def __call__(self, value):
+        return 1 if value == self._matching_value else 0
+
+    def __eq__(self, other):
+        if issubclass(type(other), IndicatorTransform):
+            return self._matching_value == other.matching_value
+        return False
+
+
+def log_transform(value):
```

Comment:
This approach is more readable, and will certainly perform better. I think we might need some comments though, as some of the transforms take numbers and others take arrays

Reply:
Agreed! I think tracking the call signatures of the transforms is prime for a class-based transform implementation. I also think checking that the arity of the transform arguments would be best to throw during runtime but before the core simulation so that the simulation doesn't crash midway.

However, I would like to keep them as simple Callables for now to improve performance and readability in master, and address this issue in a separate PR.

Reply:
That makes sense to me. Let's pull this in.